.github/contributing.md

```diff
@@ -0,0 +1,295 @@
+
+  __Unit Tests__: These are tests that focus on individual components or functions in isolation. They are typically low-level tests.
+
+  __Mid-Level Tests (Integration Tests)__: These tests focus on the integration between different components or modules. They ensure that these components work together as expected. These could be considered mid-level tests.
+
+  __End-to-End Tests__: These tests simulate real-world scenarios and interactions, typically spanning multiple components or even the entire application. They can be considered higher-level tests.
+
+
+<br>
+
+- Documentation is the last consideration that should be taken very seriously before committing code changes to the codebase.
+
+  - All types and fns where necessary should be properly documented.
+  Inherently, these comments need to follow the respective languages' comment standards.
+
+  - No unnecessary comments or `@` tags  should be visible
+
+  - Only avail of docblock syntax to further document types when language doesn't allow concise type declarations.
+
+  Comment standards
+  - PHP: `Docblock` or `Psalm`
+  - JS: `JSDoc`
+
+<br>
+
+
+__IMPORTANT__
+
+These are independent checks a contributor should ensure they've performed before requesting reviewers.
+<br>
+
+If unsure whether your changes break CI, tests or want to make a PR and have it ready, make an introductory Draft PR.
+
+<br>
+
+
+## PR creation and management
+
+To add code to our codebase after following all above steps, you *must* create a new PR for your
+changes.
+
+Integration of new code into the codebase could be achieved only by a PR towards the `dev` branch.
+
+Follow and fill in the `PULL_REQUEST_TEMPLATE` present.<br>
+Link corresponding PR to the issue at hand by adding `Closes #{ISSUE-ID}` somewhere in `Description, Changes` section to PR body or add it manually.
+
+> NOTE: Manual option is only available after PR creation from `Development` sidebar option.
+
+If unsure about whether PR is ready for review, create a draft PR.<br>
+Update PR to be full PR and ask for reviewers when ready.
+
+If changes are requested,
+1. Ask for clarification on changes necessary, if needed.
+2. Don't be afraid to state your point of view on proposed changes.
+3. Debate respectfully, acknowledge the other contributors point of view
+4. Finalize decision and solution to identify whether changes are actually necessary
+- If yes, make changes locally. <br>Follow <span style="color: #5C9340;">## Pre-committing to Github</span> guidelines defined above and push changes to Github
+5. Request re-review from contributors once ready again.
+
+<br>
+
+> General Terms:<br>
+- Every PR __must__ (if ability prohibits this) request `2` reviewers.<br>
+- Every PR __requires__ approval by at least `one` other developer.<br>
+
+  A caveat is when the PR is from a junior dev; then it must be reviewed by at least one `senior or experienced` developer.
```

Comment:
Sincerely, I don't think the distinction between `junior` and `senior` makes much sense, so I would propose to remove this line